### ISSUE TITLE
GrB_Descriptor_set is unable to set expected descriptor fields (does not match GxB_Desc_set)

### ISSUE THREAD

**Issue by eriknw** [DrTimothyAldenDavis/GraphBLAS#145]
For example, I'm unable to set `GxB_COMPRESSION` descriptor field via `GrB_Descriptor_set`, which is surprising.  I am able to set it via `GxB_Desc_set`, which I can use for now.

**Comment by DrTimothyAldenDavis** [comment]
That's by design.  GrB_Descriptor_set is limited, and can only handle settings of type GrB_Desc_Value (an enum).  But that doesn't extend to all the GxB extensions I have, some of which are double, double *, int64_t *, function pointers, and so on.  So I've limited GrB_Descriptor_set to just the GrB settings (except for two of them).  One I added earlier and it would be disruptive to remove it (the GxB_AxB_METHOD).  The other is vital for controlling the behavior of GrB_Matrix_import (the GxB_IMPORT setting).

See the definition of the GrB_Desc_Value enum in GraphBLAS.h.  It has mostly just the GrB settings.

So GxB_Desc_set, with its va_arg parameter of arbitrary type, is the one to use for GxB settings.

The GraphBLAS C API needs to provide a canonical (GrB) way to set/get settings for a descriptor, a matrix, a vector, globally, etc.  There should even be a standard way to add extensions of settings of any data type, like my GxB_Desc_set.  This is a serious limitation in the spec that I've tried to work around with GxB_Desc_set / GxB_Desc_get.

The C API Spec needs work.

In particular, for the next C API, I would suggest deleting GrB_Descriptor_set (or sun-setting it) and define new methods GrB_Desc_set and GrB_Desc_get which have both GrB parts but are also extensible.
